mock_ros2_project/service_client.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+ROS2 Service Client Example
+
+This node demonstrates how to create a service client that calls
+the AddTwoInts service.
+"""
+
+import sys
+from typing import Optional
+
+import rclpy
+from rclpy.node import Node
+from example_interfaces.srv import AddTwoInts
+
+
+class ServiceClient(Node):
+    """Service client node that calls addition service.
+    
+    This node creates a service client that sends requests to the
+    AddTwoInts service and handles the responses.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the service client node."""
+        super().__init__('service_client')
+        
+        # Create client for the AddTwoInts service
+        self.cli = self.create_client(AddTwoInts, 'add_two_ints')
+        
+        # Wait for service to be available
+        while not self.cli.wait_for_service(timeout_sec=1.0):
```

Comment:
The retry loop waits forever for the service to appear. In production nodes you may want to add a maximum number of retries or exit with a clear error so that orchestration systems (e.g., lifecycle management or launch) can react.

```suggestion
        max_retries = 5
        retry_count = 0
        while not self.cli.wait_for_service(timeout_sec=1.0):
            retry_count += 1
            self.get_logger().info('Service not available, waiting again...')
            if retry_count >= max_retries:
                self.get_logger().error(f'Service not available after {max_retries} retries. Exiting.')
                rclpy.shutdown()
                sys.exit(1)
```